rfcs/01924_tls_attestation_ca.md

```diff
@@ -0,0 +1,169 @@
+TEE application's code, since the CA only signs certificates for TEE
+applications that it attested itself.
+
+One of the main advantages of this remote attestation approach is that clients
+are able to use plain TLS for creating secure channels with TEE applications,
+and at the same time trust in this application without directly attesting it.
+
+There are several benefits from implementing _Remote Attestation via TLS_:
+
+- **The ability to run TLS applications as is**
+  - The proxy will allow clients to use TLS certificates to connect to TEE
+    applications
+  - Since attestation will be handled by the proxy - TEE applications could use
+    legacy TLS code without the need to support remote attestation logic
+- **Support multiple TEEs**
+  - Proxy could be dedicated to a specific TEE instance such as Intel SGX ECDSA,
+    AMD SEV-SNP, ARM Trustzone etc.
+    - It means that the client code wouldn’t need to support multiple
+      attestation protocols
+  - TEE applications will still be able to run on various TEE machines, since
+    the attestation logic is implemented by the proxy
+
+## Attestation
+
+This section presents a high-level attestation workflow overview. The workflow
+of the remote attestation involves 4 interacting entities:
+
+- **Certificate Authority (CA)**
+  - Remotely attests TEE applications
+  - Can be remotely attested by clients
+  - Creates signed certificates for TEE applications
+  - Distributes its root certificate among the clients
+  - [_Optional_] Can run inside the TEE environment
+- **Server**
+  - TEE application that processes requests from TLS clients
+  - Runs inside the TEE environment
+- **Client**
+  - Client application that connects to applications using TLS
+  - Runs on a user device or inside of a browser
+- **Provider**
+  - TEE platform provider that can confirm authenticity of TEE platform keys
+  - It’s important to note that the Provider is an external server (i.e. belongs
+    to AMD)
+
+It is assumed that prior to the attestation **Server** remotely attests its TEE
+platform key with the **Provider**. **Provider** sends back a signature rooted
+in some well-known root key (e.g. AMD root key), thus confirming the
+authenticity of TEE platform keys.
+
+### Attestation Workflow
+
+The detailed workflow of the remote attestation is represented by 2 stages.
+
+First stage corresponds to the backend application attestation, which happens
+only once when the backend application starts:
+
+1. **Server** generates a TLS key pair
+1. **Server** connects to the **CA**
+1. **CA** attests the **Server** using a remote attestation protocol
+   corresponding to the Server’s TEE
+   1. When requesting a TEE report from the TEE firmware, **Server** also puts
+      the hash of its TLS public key into it
+   1. For example, remote attestation could be done via
+      [#1929](https://github.com/project-oak/oak/pull/1929)
+1. **Server** sends **CA** a certificate signing request
```

Comment:
Could the server put it's CSR in the attestation token, removing a step?

Reply:
I moved it into a different step, so it wouldn't interfere with the attestation scheme that is used between the CA and the Server.